js/ui/appDisplay.js

```diff
@@ -1484,6 +1506,12 @@ const AppIcon = new Lang.Class({
         this.actor.connect('clicked', Lang.bind(this, this._onClicked));
         this.actor.connect('popup-menu', Lang.bind(this, this._onKeyboardPopupMenu));
 
+        if (iconParams['showLabel'] !== false &&
+            iconParams['editable'] === true) {
```

Comment:
It feels a bit weird to use two different syntaxes to check for positive conditions. I think you could just do
```
if (iconParams['showLabel'] && iconParams['editable'])
```